fabric-rendering-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/RenderPipelineMixin.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.rendering;
+
+import com.mojang.blaze3d.pipeline.RenderPipeline;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+
+import net.fabricmc.fabric.api.client.rendering.v1.FabricRenderPipeline;
+import net.fabricmc.fabric.impl.client.rendering.FabricRenderPipelineInternals;
+
+@Mixin(RenderPipeline.class)
+public class RenderPipelineMixin implements FabricRenderPipeline {
+	@Unique
+	boolean usePipelineDrawModeForGui = false;
+
+	@Override
+	public boolean usePipelineDrawModeForGui() {
+		return usePipelineDrawModeForGui;
+	}
+
+	static {
+		FabricRenderPipelineInternals.usePipelineDrawModeForGuiSetter = (pipeline, value) ->
```

Comment:
While this design works, I am not aware of any other places where Fabric API uses this. I would switch this to a setter method in an internal duck interface that this mixin also implements.